lib/cosmos/interfaces/interface.rb

```diff
@@ -212,6 +212,47 @@ def set_option(option_name, option_values)
       @options[option_name.upcase] = option_values.clone
     end
 
+    # Called to perform modifications on a read packet before it is inserted
+    # into the current value table.
+    #
+    # @param packet [Packet] Original packet
+    # @return [Packet] Potentially modified packet
+    def post_read_packet(packet)
+      packet
+    end
+
+    # Called to perform modifications on read data before making it into a packet
+    #
+    # @param packet_data [String] Raw packet data
+    # @return [String] Potentially modified packet data
+    def post_read_data(packet_data)
+      packet_data
+    end
+
+    # Called to perform modifications on a command packet before it is send
+    #
+    # @param packet [Packet] Original packet
+    # @return [Packet] Potentially modified packet
+    def pre_write_packet(packet)
+      packet
```

Comment:
pre_write_packet should always modify the given packet, so that the modifications get logged.  The packet might get transformed though for reasons I can't currently think of so should probably stick with returning a packet.   post_read_packet may return a different packet (the identified packet for example), so it has to return a packet